app/src/main/java/xyz/malkki/neostumbler/ui/screens/MapScreen.kt

```diff
@@ -225,12 +231,15 @@ fun MapScreen(mapViewModel: MapViewModel = viewModel()) {
 
                         //Ugly, but we don't want to update the map style unless it has actually changed
                         //TODO: think about a better way to do this
-                        if (map.style != null) {
-                            if (mapStyle.value!!.styleUrl != null && map.style!!.uri != mapStyle.value!!.styleUrl) {
-                                map.setStyle(Style.Builder().fromUri(mapStyle.value!!.styleUrl!!))
-                            } else if (mapStyle.value!!.styleJson != null && map.style!!.json != mapStyle.value!!.styleJson) {
-                                map.setStyle(Style.Builder().fromJson(mapStyle.value!!.styleJson!!))
-                            }
+                        if (
+                            map.style != null
+                            && (
+                                mapStyle.value!!.styleUrl != null && map.style!!.uri != mapStyle.value!!.styleUrl
+                                ||
+                                mapStyle.value!!.styleJson != null && map.style!!.json != mapStyle.value!!.styleJson
+                            )
+                        ) {
+                            map.setStyle(styleBuilder(mapStyle, getCoverageParams()))
```

Comment:
I think you need to also check if the coverage params have changed here. Otherwise if the coverage params have changed, the coverage won't be shown unless the map style has also changed

Yeah, it's kind of an edge case and doesn't really happen in practice, but it's better to handle it now

Reply:
When I change the settings and I back to the map the map also changed.

Reply:
That's because it's very likely that the map screen will recompose after switching tabs and the factory method will run again. But it's not guaranteed that it will recompose

Reply:
Done via https://github.com/mjaakko/NeoStumbler/compare/e833de128188e3d29cbcf9d0b0a81384b0873c55..ba63cd755621c804af990f292df3d7f8e11598cb